internal/api/admin.go

```diff
@@ -512,21 +511,18 @@ func (a *API) adminUserDelete(w http.ResponseWriter, r *http.Request) error {
 	user := getUser(ctx)
 	adminUser := getAdminUser(ctx)
 
-	var err error
+	// ShouldSoftDelete defaults to false
 	params := &adminUserDeleteParams{}
-	body, err := getBodyBytes(r)
-	if err != nil {
-		return internalServerError("Could not read body").WithInternalError(err)
-	}
-	if len(body) > 0 {
-		if err := json.Unmarshal(body, params); err != nil {
-			return badRequestError(ErrorCodeBadJSON, "Could not read params: %v", err)
+	if err := retrieveRequestParams(r, params); err != nil {
+		if err.(*HTTPError).ErrorCode == ErrorCodeBadJSON {
+			// request body is empty so the default behavior should be to hard delete the user
```

Comment:
/nit Not sure if [Unmarshal returning an error](https://github.com/supabase/auth/pull/1818/files#diff-3e8b73f5901bb9b41436eb294ebc11f27d7f3f93d07c779b0be7a771af8a3bdfR99)  is always equivalent to body is empty

Can consider removing the comment

Reply:
hmm, good point, actually i think we shouldn't ignore the other errors - will add a new error code to handle this